plugins/flutter_aepedgeidentity/test/flutter_aepedgeidentitymap_test.dart

```diff
@@ -0,0 +1,228 @@
+
+      expect(actualItemsForNamespace1Length, equals(2));
+
+      expect(actualItemsForNamespace1[0].id, expectedItemsForNamespace1[0].id);
+      expect(actualItemsForNamespace1[0].authenticatedState,
+          expectedItemsForNamespace1[0].authenticatedState);
+      expect(actualItemsForNamespace1[0].primary,
+          expectedItemsForNamespace1[0].primary);
+      expect(actualItemsForNamespace1[1].id, expectedItemsForNamespace1[1].id);
+      expect(actualItemsForNamespace1[1].authenticatedState,
+          expectedItemsForNamespace1[1].authenticatedState);
+      expect(actualItemsForNamespace1[1].primary,
+          expectedItemsForNamespace1[1].primary);
+
+      //test a non-existed namespace
+      final int actualItemsForNamespace3Length =
+          idMap.getIdentityItemsForNamespace('namespace3').length;
+      expect(actualItemsForNamespace3Length, equals(0));
+
+      //test an invalid namespace
+      final int actualItemsForNamespaceInvalidLength =
+          idMap.getIdentityItemsForNamespace('invalid').length;
+      expect(actualItemsForNamespaceInvalidLength, equals(0));
+
+      //test an empty namespace
+      final int actualItemsForNamespaceEmptyLength =
+          idMap.getIdentityItemsForNamespace('').length;
+      expect(actualItemsForNamespaceEmptyLength, equals(0));
+    });
+  });
+
+  group('IdentityMap addItem replaces items with same id', () {
+    IdentityItem item1 =
+        new IdentityItem('id1', AuthenticatedState.AMBIGUOUS, true);
+
+    IdentityItem item2 =
+        new IdentityItem('id1', AuthenticatedState.AUTHENTICATED, false);
+
+    IdentityMap idMap = new IdentityMap();
+    idMap.addItem(item1, 'namespace1');
+    idMap.addItem(item1, 'namespace1'); // same item, should be ignored
+
+    List<IdentityItem> expectedSameIdSameItem = <IdentityItem>[
+      item1,
+    ];
+
+    List<IdentityItem> expectedSameIdDiffItem = <IdentityItem>[
+      item2,
+    ];
+
+    final List<MethodCall> log = <MethodCall>[];
+
+    setUp(() {
+      channel.setMockMethodCallHandler((MethodCall methodCall) async {
+        log.add(methodCall);
+        return expectedSameIdSameItem;
+      });
+    });
+
+    test('is validated', () async {
+      final List<IdentityItem> actualSameIdSameItem =
+          idMap.getIdentityItemsForNamespace('namespace1');
+
+      //test add item with same id and same item, should ignore the new added item
+      final int actualSameIdSameItemLength =
+          idMap.getIdentityItemsForNamespace('namespace1').length;
+
+      expect(actualSameIdSameItemLength, equals(1));
+      expect(actualSameIdSameItem[0].id, expectedSameIdSameItem[0].id);
+      expect(actualSameIdSameItem[0].authenticatedState,
+          expectedSameIdSameItem[0].authenticatedState);
+      expect(
+          actualSameIdSameItem[0].primary, expectedSameIdSameItem[0].primary);
+
+      //test add item with same id but different item, should replace existing item
```

Comment:
same here, I think you can move this under its own test(...), and you will need to reinitialize idMap in the setup step which runs before each test